### ISSUE TITLE
Cluster ‘xxx’ Not Found Error When Adding Fluentd Access Log to Listener

### ISSUE THREAD

**Issue by sefaphlvn** [envoyproxy/envoy#36511]
Hi everyone,

I’m encountering an issue when trying to add a Fluentd access log to a listener in Envoy. I receive the following error:
```
[2024-10-09 18:05:42.405][75338004][warning][config] [source/extensions/config_subscription/grpc/grpc_subscription_impl.cc:138] gRPC config for type.googleapis.com/envoy.config.listener.v3.Listener rejected: Error adding/updating listener(s) testlstnrnaJDny: cluster 'abc' was not found
```

However, the Fluentd cluster abc is clearly added, as shown below:

```
{
    "dynamic_active_clusters":
    [
        {
            "cluster":
            {
                "@type": "type.googleapis.com/envoy.config.cluster.v3.Cluster",
                "alt_stat_name": "aasdaa",
                "load_assignment":
                {
                    "cluster_name": "abc",
                    "endpoints":
                    [
                        {
                            "lb_endpoints":
                            [
                                {
                                    "endpoint":
                                    {
                                        "address":
                                        {
                                            "socket_address":
                                            {
                                                "address": "1.1.1.1",
                                                "port_value": 80
                                            }
                                        }
                                    }
                                }
                            ]
                        }
                    ]
                },
                "name": "abc",
                "type": "STATIC"
            },
            "last_updated": "2024-10-09T15:05:42.392Z",
            "version_info": "12a56d24a2dca4e5778852ecdf1aa919a78e9ca35f729bdb9a7c71998cc288dc"
        }
    ]
}
```

![image](https://github.com/user-attachments/assets/833473ca-fc2a-44fa-963f-7517dda0633f)

Despite the cluster being active and correctly configured, Envoy fails to recognize it when applying the listener configuration.

Any insights or suggestions on what might be causing this discrepancy would be greatly appreciated. Thanks in advance for your help!

**Comment by alyssawilk** [comment]
yeah I'm not sure but I could imagine checkActiveStaticCluster being too narrow of a check
https://github.com/envoyproxy/envoy/blob/main/source/extensions/access_loggers/fluentd/config.cc#L42
cc @ohadvano @wbpcode

**Comment by wbpcode** [comment]
> yeah I'm not sure but I could imagine checkActiveStaticCluster being too narrow of a check https://github.com/envoyproxy/envoy/blob/main/source/extensions/access_loggers/fluentd/config.cc#L42 cc @ohadvano @wbpcode

Yeah, I think the checkActiveStaticCluster is too strict for the logger.

Hi, @sefaphlvn I think this feature require you config the cluster in the bootstrap directly. It is typically too strict for this feature.

Hi @ohadvano could you take a look, thanks?

**Comment by sefaphlvn** [comment]
Thank you for the clarification!

If configuring the cluster in the bootstrap is indeed a requirement for this feature, it would be great to mention this explicitly in the documentation to help avoid similar confusion for other users.

Thanks again for your support!

**Comment by wbpcode** [comment]
> Thank you for the clarification!
> 
> If configuring the cluster in the bootstrap is indeed a requirement for this feature, it would be great to mention this explicitly in the documentation to help avoid similar confusion for other users.
> 
> Thanks again for your support!

Yeah, I will try to release this limitation. If it's impossible, then we will update the docs. And thanks for the report. :)

**Comment by ohadvano** [comment]
> Hi @ohadvano could you take a look, thanks?

@wbpcode, I'm out for a while so can't work on a fix at this moment. If possible, please update docs that it's only supported for with static clusters. I can try see how to make it work with dynamic after I'm back.

**Comment by grnmeira** [comment]
Would this be a beginner-friendly issue? Would it be just a matter of creating a more relaxed check (disregarding the "staticness" of a cluster)? Like, just checking if it's in the active clusters collection in the cluster manager?

Given that @ohadvano is out for now, I'd be happy to give it a shot if it's a simple issue.

**Comment by Sanchit-20** [comment]
If no one is working already, I can take a stab at this, @wbpcode similar question as above, if IIUC, on the implementation front, would the additional change be beginner-friendly/simple as checking if cluster exist in dynamic cluster list or would it be something more complex?
